app/Shell/Environment.php

```diff
@@ -28,35 +26,26 @@ public function isWindowsOs(): bool
         return PHP_OS_FAMILY === 'Windows';
     }
 
-    public function portIsAvailable($port): bool
+    public function portIsAvailable(int $port): bool
     {
-        // E.g. Win/Linux: 127.0.0.1:3306 , macOS: 127.0.0.1.3306
-        $portText = $this->isLinuxOs() ? "\:{$port}\s" : "\.{$port}\s";
+        // To check if the socket is available, we'll attempt to open a socket on the port.
+        // If we cannot open the socket, it means there's nothing running on it, so the
+        // port is available. If we are successful, that means it is already in use.
+
+        $socket = @fsockopen($this->localhost(), $port, $errorCode, $errorMessage, timeout: 5);
 
-        $netstatCmd = $this->netstatCmd();
+        if (! $socket) {
+            return true;
+        }
 
-        // Check to see if the system is running a service with the desired port
-        $process = $this->shell->execQuietly("{$netstatCmd} -vanp tcp \n
-            | grep '{$portText}' | grep -v 'TIME_WAIT' | grep -v 'CLOSE_WAIT' | grep -v 'FIN_WAIT'");
+        fclose($socket);
 
-        // A successful netstat command means a port in use was found
-        return ! $process->isSuccessful();
+        return false;
     }
 
-    public function netstatCmd(): string
+    public function isTakeoutRunningOnDocker(): bool
```

Comment:
Maybe `isTakoutRunningInsideDocker`? Unless that's not a Docker-y way to say it. But `Inside` definitely makes it clearer to me; my first thought reading this was, duh, of course it is 😆